### PR TITLE
Feature/sleep data

### PR DESCRIPTION
**Pull request by Jnguyen615**
Added function for calculate a users sleep data for any given seven days. Tests to match also run the sad path.

### CODE REVIEW

File: src/model.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 export function getUserData(dataType, users, id) {
-  if (dataType === 'hydrationData' || dataType === 'sleepData') {
-    return users.filter(data => data.userID === id);
+  if (dataType === "hydrationData" || dataType === "sleepData") {
+    return users.filter((data) => data.userID === id);
   } else {
-    return users.find(data => data.id === id);
+    return users.find((data) => data.id === id);
```
Comment (reneepinna) [inline src/model.js:11]: We can remove these extra parens.

File: src/model.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ function getRandomID(array) {
 }
 
 export function getRandomUser(users) {
-  return getUserData('users', users, getRandomID(users));
+  return getUserData("users", users, getRandomID(users));
```
Comment (reneepinna) [inline src/model.js]: We can go ahead and swap this to a single quote.

File: src/model.js [hunk 3]
```diff
@@ -61,31 +61,40 @@ export function getWeeklyWater(userData) {
...
     userData.reduce((acc, user) => (acc += user.sleepQuality), 0) /
-    userData.length
-  );
+    userData.length;
+  return parseFloat(avg.toFixed(1));
 }
 
 export function getDailySleep(sleepData, data) {
-  const userData = sleepData.find(date => data === date.date);
+  const userData = sleepData.find((date) => data === date.date);
```
Comment (reneepinna) [inline src/model.js]: We can remove the parens here.

File: src/model.js [hunk 4]
```diff
@@ -61,31 +61,40 @@ export function getWeeklyWater(userData) {
...
 }
 
 export function getSleepQuality(sleepData, date) {
-  const userData = sleepData.find(data => data.date === date)
+  const userData = sleepData.find((data) => data.date === date);
+  return userData.sleepQuality;
+}
+
+export function getWeeklySleep(userData, date) {
+  const startCount = userData.findIndex((entry) => entry.date === date)
```
Comment (reneepinna) [inline src/model.js]: We can remove the parens here.

**Review by reneepinna — CHANGES_REQUESTED** [review]
Fabulous job, just a few formatting things to address before we merge in.

**Review by JoshBennett793 — CHANGES_REQUESTED** [review]
Can we reformat everything back to single quotes in the sampleData file as well?

**Review by aliceabarca — COMMENTED** [review]
Looks good!

**Pull request merged**